### PR TITLE
update OSO endpoint and queries

### PR DESCRIPTION
**Pull request by MasterHW**
Fixes: https://github.com/gitcoinco/grants-stack/issues/3669

## Description

https://linear.app/gitcoin/issue/PAR-438/update-oso-endpoint-and-queries
With current OSO models it is not straightforward to retrieve code velocity data. Updated that field to instead reflect OSO's `activeDeveloperCount6Months` field for a sense of active developers.

## Checklist

This PR:

- [ ] Does it add new payout or donation token? In this case, have they been added to the indexer to avoid it to crash?
- [ ] Includes only changes relevant to the original ticket. Significant refactoring needs to be separated.
- [ ] Doesn't disable eslint rules.
- [ ] Doesn't work around the type checker (including but not limited to: type casts, non-null assertions, `@ts-ignore`, unjustified optional values).
- [ ] Doesn't contain commented out code.
- [ ] If adding/updating a feature, it adds/updates its test script on Notion.

### CODE REVIEW

**Comment by MasterHW** [comment]
known issue: when browsing through applications in explorer, Impact Stats will not populate the page after visiting a project with no stats (even if the application you're looking at does have them). page refresh fixes this. caching/cookie issue?

**Comment by MasterHW** [comment]
> known issue: when browsing through applications in explorer, Impact Stats will not populate the page after visiting a project with no stats (even if the application you're looking at does have them). page refresh fixes this. caching/cookie issue?

believe this is fixed. improved logic determining when stats are queried to account for circumstances where stats have already been queried, but not for the currently viewed project.

File: packages/grant-explorer/src/features/api/oso.ts [hunk 1]
```diff
@@ -4,147 +4,70 @@ import { Hex } from "viem";
...
-      const idData: IOSOId = await graphQLClient.request<IOSOId>(queryId);
+      const items: IOSOStats = await graphQLClient.request<IOSOStats>(
+        queryStats,
+        queryVars
+      );
 
-      if (!Array.isArray(idData.projects_v1)) {
-        setStats(emptyReturn);
-        return;
+      if (!Array.isArray(items.oso_codeMetricsByProjectV1)) {
```
Comment (veganbeef) [inline packages/grant-explorer/src/features/api/oso.ts:60]: You could combine this if statement with the else condition on line 70 with something like:
```
if (!items.oso_codeMetricsByProjectV1?.length) {
  throw new Error("no stats returned");
}
```
and then the `const parsedItems` block can be pulled out of that if condition and you don't need to duplicate the error.

That more closely follows the "return early" pattern, which makes things easier to read IMO.

**Review by veganbeef — APPROVED** [review]
LGTM, just one style suggestion

File: packages/grant-explorer/src/features/api/oso.ts (hunk 1 above)
Comment (MasterHW) [reply packages/grant-explorer/src/features/api/oso.ts:60]: cheers, will incorporate!

**Review by gnomadic — APPROVED** [review]

**Pull request merged**